tests/steamship_tests/agents/test_agent_service.py

```diff
@@ -248,3 +248,38 @@ def test_context_logging_to_chat_history_everything(client: Steamship):
         assert not has_status_message(chat_history.messages, RoleTag.AGENT)
         assert not has_status_message(chat_history.messages, RoleTag.LLM)
         assert has_status_message(chat_history.messages, RoleTag.TOOL)
+
+
+@pytest.mark.usefixtures("client")
+def test_async_prompt(client: Steamship):
+    example_agent_service_path = (
+        SRC_PATH / "steamship" / "agents" / "examples" / "example_assistant.py"
+    )
+    with deploy_package(client, example_agent_service_path, wait_for_init=True) as (
+        _,
+        _,
+        agent_service,
+    ):
+        context_id = "some_async_fun"
+        streaming_resp = agent_service.invoke(
+            "async_prompt",
+            prompt="who is the current president of the United States?",
+            context_id=context_id,
+        )
+
+        assert streaming_resp is not None
+        assert streaming_resp["file"] is not None
+        assert streaming_resp["task"] is not None
+
+        file = File(client=client, **(streaming_resp["file"]))
+        task = Task(client=client, **(streaming_resp["task"]))
+
+        original_len = len(file.blocks)
+
+        task.wait()
```

Comment:
The fact that we can still wait on the task is 💯 .

Reply:
@dkolas it is complicated. The _existing_ LLM `chat()` call waits on the full `generate()` at the moment. This may change when we add `streaming=True`, but I _think_ we'll want to make that have a way to wait on all the returned blocks to finish streaming (at least, something in the `Agent` must, as the next action must be selected).

So, at the moment, that `task` is waiting on everything. And, it _feels_ like it will always do that. To stream, the client would ignore it and watch the `File` events (which I thought was the way we wanted it).